pandas/core/generic.py

```diff
@@ -9999,6 +9999,9 @@ def mask(
         cond = common.apply_if_callable(cond, self)
         other = common.apply_if_callable(other, self)
 
+        if isinstance(cond, (ABCDataFrame, ABCSeries)):
```

Comment:
Rather than a fillna like this do you see where the logic is breaking down that causes `.mask` to treat `pd.NA` the same was as `True`? We shouldn't have to mess around with pd.NA equality semantics for this fix

Reply:
For indexing this goes through `check_array_indexer` which essentially also converts NA to False (but lower in the stack):

https://github.com/pandas-dev/pandas/blob/2aa155ae1cf019e902441463d1af873eaa3c803b/pandas/core/indexers/utils.py#L532-L534